### PR TITLE
[DO NOT MERGE] feat: Add OIDC support with ID-porten

### PR DESCRIPTION
**Pull request by netroms**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/oidc/DhisClientRegistrationRepository.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            .authorizationGrantType( AuthorizationGrantType.AUTHORIZATION_CODE )
+            .redirectUriTemplate( redirBaseUri + "/login/oauth2/code/{registrationId}" )
+            .scope( "openid", "profile" )
+            .authorizationUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/authorize" )
+            .tokenUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/token" )
+            .userInfoUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/userinfo" )
+            .userNameAttributeName( IdTokenClaimNames.SUB )
+            .userInfoAuthenticationMethod( AuthenticationMethod.HEADER )
+            .jwkSetUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/jwk" )
+            .clientName( "Google" )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/oidc/DhisClientRegistrationRepository.java:82]: Is "Google" correct here?

Comment (netroms) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/oidc/DhisClientRegistrationRepository.java:82]: Nope, det er glipp, ser ut til å fungere likevel

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java [hunk 2]
```diff
@@ -1,714 +1,721 @@
...
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java:14]: Do you know why we get a full diff on this class?

Comment (netroms) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java:14]: CRLF issue

**Pull request closed without merging**